reef-runtime-mesos/src/main/java/org/apache/reef/runtime/mesos/driver/MesosScheduler.java

```diff
@@ -0,0 +1,501 @@
+        this.mesosRemoteManager.getHandler(taskStatus.getMessage(), EvaluatorReleaseProto.class);
+    this.executors.add(taskStatus.getTaskId().getValue(), resourceRequestProto.getMemorySize(),
+        evaluatorLaunchHandler, evaluatorReleaseHandler);
+
+    final ResourceAllocationProto alloc = DriverRuntimeProtocol.ResourceAllocationProto.newBuilder()
+        .setIdentifier(taskStatus.getTaskId().getValue())
+        .setNodeId(taskStatus.getSlaveId().getValue())
+        .setResourceMemory(resourceRequestProto.getMemorySize())
+        .build();
+    reefEventHandlers.onResourceAllocation(alloc);
+
+    this.outstandingRequestCounter--;
+    this.updateRuntimeStatus();
+  }
+
+  private synchronized void updateRuntimeStatus() {
+    final Builder builder = DriverRuntimeProtocol.RuntimeStatusProto.newBuilder()
+        .setName(RUNTIME_NAME)
+        .setState(State.RUNNING)
+        .setOutstandingContainerRequests(this.outstandingRequestCounter);
+
+    for (final String executorId : this.executors.getExecutorIds()) {
+      builder.addContainerAllocation(executorId);
+    }
+
+    this.reefEventHandlers.onRuntimeStatus(builder.build());
+  }
+
+  private void onRuntimeError(final Throwable throwable) {
+    this.executorService.shutdown();
+    this.mesosMaster.stop();
+
+    final Builder runtimeStatusBuilder = RuntimeStatusProto.newBuilder()
+        .setState(State.FAILED)
+        .setName(RUNTIME_NAME);
+
+    final Encoder<Throwable> codec = new ObjectSerializableCodec<>();
+    runtimeStatusBuilder.setError(ReefServiceProtos.RuntimeErrorProto.newBuilder()
+        .setName(RUNTIME_NAME)
+        .setMessage(throwable.getMessage())
+        .setException(ByteString.copyFrom(codec.encode(throwable)))
+        .build());
+
+    this.reefEventHandlers.onRuntimeStatus(runtimeStatusBuilder.build());
+  }
+
+  private boolean satisfySlaveConstraint(final ResourceRequestProto resourceRequestProto, final Offer offer) {
+    return resourceRequestProto.getNodeNameCount() == 0 ||
+        resourceRequestProto.getNodeNameList().contains(offer.getSlaveId().getValue());
+  }
+
+  private int getMemory(final Offer offer) {
+    for (final Resource resource : offer.getResourcesList()) {
+      switch (resource.getName()) {
+        case "mem":
+          return (int)resource.getScalar().getValue();
+      }
+    }
+    throw new IllegalArgumentException("No memory specified in offer");
+  }
+
+  private int getCpu(final Offer offer) {
+    for (final Resource resource : offer.getResourcesList()) {
+      switch (resource.getName()) {
+        case "cpus":
+          return (int)resource.getScalar().getValue();
+      }
+    }
+    throw new IllegalArgumentException("No cpu specified in offer");
+  }
+
+  private String getExecutorLaunchCommand(final String executorID, final int memorySize) {
+    final String DEFAULT_JAVA_PATH = System.getenv("JAVA_HOME") + "/bin/" +  "java";
```

Comment:
http://mesos.apache.org/gettingstarted/

Under the "System Requirements" section, it says "Mesos runs on Linux (64 Bit) and Mac OSX (64 Bit)."